R/is_valid_cusip.R

```diff
@@ -0,0 +1,56 @@
+#' Validate a vector of CUSIPs
+#'
+#' This function validates that a vector of CUSIPs are valid CUSIP codes.
+#'
+#' @param cusips A character or numeric vector (automatically collapses a single
+#'   column data.frame to a vector)
+#'
+#' @return A logical vector the same length as `cusips`.
```

Comment:
Although, in that case, maybe `PACTA.analysis` wouldn't be the home for it. 

Ok on second thought, it is probably fine as is.

Reply:
I don't like the idea of changing the output. I think it should always return an object of the same type and dimension, and it should be the most minimal object possible, e.g. a logical vector not a data frame. We could throw warnings every time something doesn't match, but since this is primarily intended to be run on a large vector, that could easily end up more trouble than it's worth. Maybe a separate utility function could be made for that?

Reply:
Ya that makes sense. It might make sense at one point to sit down and think about our ethos for `PACTA.analysis`, and the code therein. 

For `r2dii`, our ethos was definitely to focus on how user-friendly it was, and tried to integrate it seamlessly with `tidyverse`. For `PACTA.analysis`, it seems the focus should be more on speed, maintainability, etc. since users may not use it directly as much. 

Anyway, this LGTM, merge whenever.